src/net/tcp.rs

```diff
@@ -119,7 +121,16 @@ impl TcpStream {
     /// it should already be connected via some other means (be it manually, the
     /// net2 crate, or the standard library).
     pub fn from_stream(stream: net::TcpStream) -> io::Result<TcpStream> {
-        try!(stream.set_nonblocking(true));
+
+        #[cfg(not(target_os = "fuchsia"))]
+        { try!(stream.set_nonblocking(true)); }
+
+        #[cfg(target_os = "fuchsia")]
+        {
+            try!(sys::set_nonblock(
+                ::std::os::unix::io::AsRawFd::as_raw_fd(&stream)));
```

Comment:
Oh sorry forgot to comment on this earlier, but could this logic be pushed into the fuchsia module? Ideally this module would have no `#[cfg]` in theory